### ISSUE TITLE
MSSQL Linked Servers support

### ISSUE THREAD

**Issue by panuckov-nick** [linq2db/linq2db#1125]
Hi,

I walk through documentation and didn't found any information about support of the Linked Servers in MSSQL.
Only thing that I found and using is extension method DatabaseName that is generating -SQL like this:
```sql
FROM [MyDatabase].[dbo].[User] 
```
that is completely ok while query consist of tables located on one server. But when I need join few tables located on different servers and each server has link to each other via MSSQL Linked Server object I would expect similar method that will allow me to define specific linked server name like this:
```cs
DatabaseName("MyDatabase").ServerName("MyServer")
```
that will generate T-SQL query like this:
```sql
FROM [MyServer].[MyDatabase].[dbo].[User]
```

**Comment by MaceWindu** [comment]
We also need to check other servers. At least Informix supports it too if I remember properly

**Comment by sdanyliv** [comment]
@MaceWindu, how do you think, it is better to add such extension methods for ITable?

**Comment by MaceWindu** [comment]
We should add extra server parameter/method everywhere where we have db/schema/table parameters and methods

**Comment by MaceWindu** [comment]
I will try to find time for this this or next week so it will get into 2.0, as this change could be API breaking, and not suitable for 2.x release

**Comment by sdanyliv** [comment]
It's not breaking API, just extending.

**Comment by MaceWindu** [comment]
Moving it to 2.1 milestone as it will have too much changes to accept it for 2.0 for now, which is in code-freeze already.

**Comment by keaton-tagseth** [comment]
Has this been addressed at all? It was moved into 2.2.0 but i haven't seen anything indicating it was actually implemented in the 2.2.0 release.

**Comment by MaceWindu** [comment]
No, it was postponed, but I will add it to next milestone as I see there is an interest in it

**Comment by sdanyliv** [comment]
@keaton-tagseth, there is working workaround if you enclose server name in brackets:
```cs
 db.GetTable<Some>().DatabaseName("[ServerName].[DatabaseName]")
```

**Comment by MaceWindu** [comment]
closed by #1357
